lib/ixml.js

```diff
@@ -173,9 +174,21 @@ const iXmlNodePgmClose = () => {
   return I_XML_NODE_PGM_CLOSE;
 }
 
-const iXmlNodeParmOpen = (xio) => {
+const iXmlNodeParmOpen = (opt) => {
+  if (!(opt && typeof opt==='object')){
+    return iXmlNodeOpen(I_XML_NODE_PARM_OPEN)
+    + I_XML_NODE_CLOSE;
+  }
+
+  let opt_io=(opt.io) ? iXmlAttrDefault(I_XML_ATTR_KEY_IO,opt.io,I_XML_ATTR_VALUE_OPTIONAL) : "";
+  let opt_by=(opt.by) ? iXmlAttrDefault(I_XML_ATTR_KEY_BY,opt.by,I_XML_ATTR_VALUE_OPTIONAL) : "";
+
+  delete opt.by;
+  delete opt.io;

```

Comment:
Why are we deleting values from the user's object? Especially, since we don't pass opt to anything after this point.

Reply:
It looks like itoolkit.js uses it as an input to iXmlNodeDataOpen.  I figured if they weren't going to be used there, it would be best to eliminate.